src/forms/TextInput.jsx

```diff
@@ -5,18 +5,14 @@ import cx from 'classnames';
 /**
  * @module TextInput
  */
-class TextInput extends React.Component {
+class TextInput extends React.PureComponent {
 
 	constructor(props) {
 		super(props);
-		this.state = {
-			value: props.value || '',
-		};
 		this.onChange = this.onChange.bind(this);
 	}
 
 	onChange(e) {
-		this.setState({ value: e.target.value });
 		if (this.props.onChange) {
 			this.props.onChange(e);
```

Comment:
If you attached this directly to the [onChange prop](https://github.com/meetup/meetup-web-components/pull/246/files#diff-ec0100a5c023d94b8141bd5e7c37d91eR64) of the input, you could really trim down a lot of the code in this component:

```jsx
<input onChange={this.props.onChange} ... />
```

You'd be able to delete the `constructor` and `onChange` methods altogether while keeping the same functionality. You could then even go one step further and define this as a stateless functional  component instead of a class if you wanted (no need to worry about `this` or lifecycle methods):

```jsx
class TextInput extends React.PureComponent {
  render() {
    const { /* destructure props */ } = this.props;
    return <input ... />
  }
}

// vs

function TextInput({/* destructure props */}) {
  return <input ... />
}
```